src/ui/src/composables/usePromiseQueue.ts

```
@@ -0,0 +1,62 @@
+					const data = await func();
+					res(data);
+				} catch (e) {
+					rej(e);
+					throw e;
+				}
+			}
+
+			const job = { func: callback, id: incrementNextId() };
+			jobs.push(job);
+			await run();
+		});
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)